tests/NHSD.BuyingCatalogue.Solutions.Persistence.DatabaseTests/AdditionalServiceRepositoryTests.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using NHSD.BuyingCatalogue.Solutions.Contracts.Persistence;
+using NHSD.BuyingCatalogue.Testing.Data;
+using NHSD.BuyingCatalogue.Testing.Data.EntityBuilders;
+using NUnit.Framework;
+
+namespace NHSD.BuyingCatalogue.Solutions.Persistence.DatabaseTests
+{
+    [TestFixture]
+    internal sealed class AdditionalServiceRepositoryTests
+    {
+        private IAdditionalServiceRepository _additionalServiceRepository;
+
+        private const string SupplierId = "Sup1";
+
+        [SetUp]
+        public async Task SetUp()
+        {
+            await Database.ClearAsync();
+
+            await SupplierEntityBuilder.Create()
+                .WithId(SupplierId)
+                .Build()
+                .InsertAsync();
+
+            TestContext testContext = new TestContext();
+            _additionalServiceRepository = testContext.AdditionalServiceRepository;
+        }
+
+        [Test]
+        public async Task GetAdditionalServiceByCatalogueItemIdAsync_InvalidSolutionId_ReturnsEmptyList()
+        {
+            var request =
+                await _additionalServiceRepository.GetAdditionalServiceBySolutionIdsAsync(
+                    new List<string> { "INVALID" }, CancellationToken.None);
+
+            request.Count().Should().Be(0);
+        }
+
+        [Test]
+        public async Task GetAdditionalServiceByCatalogueItemIdAsync_SingleSolutionId_ReturnsOneAdditionalService()
+        {
+            const string solutionId = "Sln1";
+
+            await CreateSolution(solutionId);
+            await CreateAdditionalService("Cat", solutionId);
+
+            var request =
+                await _additionalServiceRepository.GetAdditionalServiceBySolutionIdsAsync(
+                    new List<string> { solutionId }, CancellationToken.None);
+
+            request.Count().Should().Be(1);
```

Comment:
We should probably verify the content of the list as well as the count

Reply:
👍